numpy/core/fromnumeric.py

```diff
@@ -62,6 +61,32 @@ def _wrapfunc(obj, method, *args, **kwds):
         return _wrapit(obj, method, *args, **kwds)
 
 
+def _wrapreduction(obj, method, name, axis, dtype, out, **kwargs):
+    passkwargs = {}
+    for k, v in kwargs.items():
+        if v is not np._NoValue:
+            passkwargs[k] = v
+
+    if isinstance(obj, _gentype):
+        res = reduce(method, obj, method.identity)
+        if out is not None:
```

Comment:
In my opinion we want to deprecate this, not add support for it to the other function:

* It's not documented
* It completely ignores the `axis`, `dtype`, and `keepdims` arguments
* There's no precedent for accepting generators elsewhere in numpy

Reply:
I'll remove it completely. Hopefully no tests fail.